data/txHandler.go

```diff
@@ -582,8 +576,11 @@ func (handler *TxHandler) processIncomingTxn(rawmsg network.IncomingMessage) net
 		// if the elastic rate limiter cannot vend a capacity, the error it returns
 		// is sufficient to indicate that we should enable Congestion Control, because
 		// an issue in vending capacity indicates the underlying resource (TXBacklog) is full
-		capguard, err = handler.erl.ConsumeCapacity(rawmsg.Sender.(util.ErlClient))
-		if err != nil {
+		capguard, err = handler.erl.ConsumeCapacity(rawmsg.Sender)
+		// do not enable congestion control if consumption failed because the sender couldn't be used for consumption
+		// as that sender is simply unable to consume capacity (and therefore should not affect congestion control)
+		// this should not currently be possible as all senders are wsPeers
+		if err != nil && !errors.Is(err, util.ErrIncompatibleERLClient) {
```

Comment:
Is it possible to proceed to call `handler.erl.EnableCongestionControl` for a specific error type returned, instead of any error excep `util.ErrIncompatibleERLClient`?

Reply:
Yep! I wanted to make sure I got this to you early so I didn't do too much messing with error values, but I agree that'd be better.

I think what I'll do is wrap these errors into two kinds of errors: "CapacityFault" and "Other" (with better names) -- capacity fault indicating that capacity couldn't be vended, and "other" indicating that something unexpected/unrelated happened.

Then this check can be a positive check `errors.Is(err, util.ERLCapacityFault)` instead of a negative one